### PR TITLE
Added Sourcegraph badge

### PR DESCRIPTION
**Pull request by rohanpai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [![Gitter chat](http://img.shields.io/badge/gitter-join%20chat%20%E2%86%92-brightgreen.svg)](https://gitter.im/opentracing/public) [![Build Status](https://travis-ci.org/opentracing/opentracing-go.svg?branch=master)](https://travis-ci.org/opentracing/opentracing-go) [![GoDoc](https://godoc.org/github.com/opentracing/opentracing-go?status.svg)](http://godoc.org/github.com/opentracing/opentracing-go)
+[![Sourcegraph Badge](https://sourcegraph.com/github.com/opentracing/opentracing-go/-/badge.svg)](https://sourcegraph.com/github.com/opentracing/opentracing-go?badge)
```
Comment (yurishkuro) [inline README.md]: The badge says "used by N projects", but the link just shows the source. Can sourcegraph actually show those N projects?

Comment (rohanpai) [inline README.md]: Hi @yurishkuro , that's currently in the works Once we add that in, the repo's landing page will show that data!

Comment (bhs) [inline README.md]: I would also like to see ^^^ happen. Can SG provide a rough time estimate? At the moment this is cool but not that useful (in terms of finding working examples, etc).

Comment (bhs) [inline README.md]: (ping, @rohanpai)

**Comment by tedsuo** [comment]
@rohanpai @yurishkuro improvements to sourcegraph aside,  this LGTM. Can this be merged in?

**Comment by yurishkuro** [comment]
it feels misleading without the actual functionality the badge promises. Without it it's just a link to the source code.

**Comment by slimsag** [comment]
(Sourcegraph dev here) So sorry this got lost track of, @bhs @yurishkuro @tedsuo  ! -- @rohanpai served as an intern for us when he sent this PR but this must've fell through the cracks when he headed back to school.

We recently added this functionality. Clicking on badges now open a panel on Sourcegraph.com to show the actual project users, for example, clicking on opentracing-go badge here:

[Sourcegraph Badge](https://sourcegraph.com/github.com/opentracing/opentracing-go?badge)

---

</img>

---

Clicking each of these will also bring you to that repository, searching for the import path to see where the usage is:

---

</img>

---

I'd be glad to hear more feedback if this functionality doesn't seem useful enough! And of course, feel absolutely free to close/deny this PR if you think it's not a good change :)

**Comment by yurishkuro** [comment]
looks great!

**Pull request merged**